# Assignment for Tools and Programming

This Assignment was created by:
Martin Brucker and Frederik Brinkmann

Deadline: 26.10.2023 11:59PM

In [2]:
import pandas as pd
import numpy as np

In [3]:
# importing data from files

results = pd.read_csv("data/results.csv")
athletes = pd.read_csv(filepath_or_buffer="data/athletes.txt", sep=";")

In [4]:
merged_results_athletes = pd.merge(left = athletes, right = results, how="inner")

#### Exercise 1
- What is the name of the most successful athlete since 2012, as measured by the number of gold
medals?

In [6]:
gold_medals_df = merged_results_athletes[merged_results_athletes['medal'] == 'Gold']
gold_medals_count = gold_medals_df['name'].value_counts()
print(gold_medals_count)

name
Michael•Phelps           23
Carl•Lewis                9
Paavo•Nurmi               9
Mark•Spitz                9
Larisa•Latynina           9
                         ..
Heinrich•Schomburgk       1
Michael•Stich             1
Fritz•Traun               1
Paige•Hauschild           1
Stephania•Haralabidis     1
Name: count, Length: 10874, dtype: int64


As we can see, Michael Phelps won the most gold Medals with a count of 23

- How many different athletes have participates in the Olympic Games since 2012?

In [10]:
since2012 = merged_results_athletes[merged_results_athletes['year'] >= 2012]
since2012.nunique()

athlete         29655
name            29518
sex                 2
born            27730
died               58
affiliations    12526
title               7
measurements     2698
year                5
season              2
sport              64
discipline        382
pos               139
medal               3
country           210
country_code      210
team             3910
dtype: int64

as we can see there are 29655 unquie athleets since 2012

- What are the 10 countries with the lowest ratio of female participants? Represent this information using a bar chart